src/EntityRegistrantDetection/EntityRegistrantDetectionManager.php

```diff
@@ -0,0 +1,270 @@
+						'hierarchical' => $post_type->hierarchical,
+						'source'       => $source,
+					];
+					break;
+				case 'taxonomy':
+					$taxonomy = get_taxonomy( $entity );
+
+					if ( empty( $taxonomy ) || ! $taxonomy instanceof WP_Taxonomy ) {
+						break;
+					}
+
+					self::$taxonomies_source[ $entity ] = [
+						'name'         => $taxonomy->label,
+						'slug'         => $taxonomy->name,
+						'description'  => $taxonomy->description,
+						'hierarchical' => $taxonomy->hierarchical,
+						'source'       => $source,
+					];
+					break;
+				case 'block':
+					$block_types = WP_Block_Type_Registry::get_instance()->get_all_registered();
+
+					if ( empty( $block_types[ $entity ] ) || ! $block_types[ $entity ] instanceof WP_Block_Type ) {
+						break;
+					}
+
+					$block_type = $block_types[ $entity ];
+
+					self::$blocks_source[ $entity ] = [
+						'name'        => $block_type->name,
+						'title'       => $block_type->name,
+						'description' => $block_type->description,
+						'category'    => $block_type->category,
+						'attributes'  => $block_type->get_attributes(),
+						'is_dynamic'  => $block_type->is_dynamic(),
+						'source'      => $source,
+					];
+					break;
+				case 'shortcode':
+					global $shortcode_tags;
+
+					if ( ! isset( $shortcode_tags[ $entity ] ) ) {
+						break;
+					}
+
+					self::$shortcodes_source[ $entity ] = [
+						'tag'    => $entity,
+						'source' => $source,
+					];
+					break;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Adds hooks.
+	 */
+	public function register() {
+
+		$int_min = defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : ~PHP_INT_MAX; // phpcs:ignore PHPCompatibility.Constants.NewConstants.php_int_minFound
+
+		add_action( 'all', [ $this, 'wrap_hook_callbacks' ], $int_min );
+	}
+
+	/**
+	 * Wrap action/filter callback to given hook to check if callback registering any entity or not.
+	 *
+	 * @global WP_Hook[] $wp_filter
+	 *
+	 * @param string $hook Hook name for action or filter.
+	 *
+	 * @return void
+	 */
+	public function wrap_hook_callbacks( $hook ) {
+
+		global $wp_filter;
+
+		if ( ! isset( $wp_filter[ $hook ] ) || 'shutdown' === $hook ) {
+			return;
+		}
+
+		foreach ( $wp_filter[ $hook ]->callbacks as $priority => &$callbacks ) {
+			foreach ( $callbacks as &$callback ) {
+				if ( ! $callback['function'] instanceof CallbackWrapper ) {
+					$callback['function'] = self::wrapped_callback(
+						array_merge(
+							$callback,
+							compact( 'priority', 'hook' )
+						)
+					);
+				}
+			}
+		}
+
+	}
+
+	/**
+	 * Wrap the given callable function.
+	 *
+	 * @param array $callback [
+	 *     The callback data.
+	 *     @type callable $function
+	 *     @type int      $accepted_args
+	 *     @type int      $priority
+	 *     @type int      $hook
+	 * ]
+	 *
+	 * @return CallbackWrapper Instance of CallbackWrapper.
+	 */
+	protected static function wrapped_callback( $callback ) {
+
+		return new CallbackWrapper( $callback );
```

Comment:
This could then do:

```suggestion
		$wrapper = $this->injector->make( CallbackWrapper::class );
		$wrapper->set_callback( $callback );
		return $wrapper;
```

Maybe `$callback` could still be supplied among the `$arguments` of the `make` method too.